src/QMCWaveFunctions/tests/test_J1OrbitalSoA.cpp

```diff
@@ -62,6 +61,108 @@ TEST_CASE("J1 evaluate derivatives Jastrow", "[wavefunction]")
+  twf.evaluateLog(elec_);
+  twf.prepareGroup(elec_, 0);
+
+  auto& twf_component_list = twf.getOrbitals();
+
+  opt_variables_type active;
+  twf.checkInVariables(active);
+
+  int nparam = active.size_of_active();
+  REQUIRE(nparam == 4);
+
+  using ValueType = QMCTraits::ValueType;
+  std::vector<ValueType> dlogpsi(nparam);
+  std::vector<ValueType> dhpsioverpsi(nparam);
+  //twf.evaluateDerivatives(elec_, active, dlogpsi, dhpsioverpsi);
+  twf_component_list[0]->evaluateDerivatives(elec_, active, dlogpsi, dhpsioverpsi);
+
+  // Numbers not validated
+  std::vector<ValueType> expected_dlogpsi      = {-0.9336294487, -1.0196051794, 0.0, 0.0};
+  std::vector<ValueType> expected_dhpsioverpsi = {-1.1596433096, 0.7595492539, 0.0, 0.0};
+  for (int i = 0; i < nparam; i++)
+  {
+    CHECK(dlogpsi[i] == ValueApprox(expected_dlogpsi[i]));
+    CHECK(dhpsioverpsi[i] == ValueApprox(expected_dhpsioverpsi[i]));
+  }
+}
+
+TEST_CASE("J1 evaluate derivatives Jastrow with missing J1 for group", "[wavefunction]")
+{
+  Communicate* c = OHMMS::Controller;
+  auto ions_uptr = std::make_unique<ParticleSet>();
+  auto elec_uptr = std::make_unique<ParticleSet>();
+  ParticleSet& ions_(*ions_uptr);
+  ParticleSet& elec_(*elec_uptr);
+
+  ions_.setName("ion0");
+  ions_.create({1, 1});
+  ions_.R[0]                 = {0.0, 0.0, 1.0};
+  ions_.R[1]                 = {0.0, 0.0, 0.0};
+  SpeciesSet& ispecies       = ions_.getSpeciesSet();
+  int OIdx                   = ispecies.addSpecies("O");
+  int HIdx                   = ispecies.addSpecies("H");
+  int imassIdx               = ispecies.addAttribute("mass");
+  int ichargeIdx             = ispecies.addAttribute("charge");
+  ispecies(ichargeIdx, HIdx) = 1.0;
+  ispecies(ichargeIdx, OIdx) = 8.0;
+  ispecies(imassIdx, HIdx)   = 1.0;
+  ispecies(imassIdx, OIdx)   = 16.0;
+
+  elec_.setName("e");
+  elec_.create({1, 1});
+  elec_.R[0] = {0.5, 0.5, 0.5};
+  elec_.R[1] = {-0.5, -0.5, -0.5};
+
+  SpeciesSet& tspecies       = elec_.getSpeciesSet();
+  int upIdx                  = tspecies.addSpecies("u");
+  int downIdx                = tspecies.addSpecies("d");
+  int massIdx                = tspecies.addAttribute("mass");
+  int chargeIdx              = tspecies.addAttribute("charge");
+  tspecies(massIdx, upIdx)   = 1.0;
+  tspecies(massIdx, downIdx) = 1.0;
+  tspecies(chargeIdx, upIdx) = -1.0;
+  tspecies(massIdx, downIdx) = -1.0;
+  // Necessary to set mass
+  elec_.resetGroups();
+
+  ParticleSetPool ptcl = ParticleSetPool(c);
+  ptcl.addParticleSet(std::move(elec_uptr));
+  ptcl.addParticleSet(std::move(ions_uptr));
+
+
+  ions_.update();
+  elec_.addTable(elec_);
+  elec_.addTable(ions_);
+  elec_.update();
+
+  ions_.get(app_log());
+  elec_.get(app_log());
+
   const char* jasxml = "<wavefunction name=\"psi0\" target=\"e\"> \
   <jastrow name=\"J1\" type=\"One-Body\" function=\"Bspline\" print=\"yes\" source=\"ion0\"> \
     <correlation elementType=\"H\" cusp=\"0.0\" size=\"2\" rcut=\"5.0\"> \
```

Comment:
@shivupa Missing ion group in J1 seems not fully supported. The added test was causing problems, so I add J1O parameter to make it passing for now. We definitely need to fix the case of missing group but not in this PR and also not adding two dummy parameters as the added test indicates.

Reply:
Ok I can take a look in a following PR